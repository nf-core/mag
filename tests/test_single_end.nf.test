nextflow_pipeline {

    name "Test pipeline with test_single_end profile"
    script "../main.nf"
    tag "pipeline"
    tag "test_single_end"
    profile "test_single_end"

    test("-profile test_single_end") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // Output directory: multiqc
            def stable_name_multiqc = getAllFilesFromDir(
                params.outdir, include: ['multiqc/**'], relative: true, includeDir: false
            )
            def stable_path_multiqc = getAllFilesFromDir(
                params.outdir, include: ['multiqc/**'], ignoreFile: 'tests/.nftignore'
            )
            // // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // // stable_path: All files in ${params.outdir}/ with stable content
            // def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_mag_software_mqc_versions.yml"),
                    // // All stable path name, with a relative path
                    // stable_name,
                    // // All files with stable contents
                    // stable_path
                ).match() },
                { assert snapshot(stable_name_multiqc, stable_path_multiqc).match('multiqc') },

            )
        }
    }
}
