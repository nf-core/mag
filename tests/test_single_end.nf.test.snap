{
    "-profile test_single_end": {
        "content": [
            157,
            {
                "ADAPTERREMOVAL_SE": {
                    "adapterremoval": "2.3.2"
                },
                "ADJUST_MAXBIN2_EXT": {
                    "coreutils": null
                },
                "BCFTOOLS_CONSENSUS": {
                    "bcftools": 1.17
                },
                "BCFTOOLS_INDEX_POST": {
                    "bcftools": 1.17
                },
                "BCFTOOLS_INDEX_PRE": {
                    "bcftools": 1.17
                },
                "BCFTOOLS_VIEW": {
                    "bcftools": 1.17
                },
                "BIN_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "BOWTIE2_ASSEMBLY_ALIGN": {
                    "bowtie2": "2.4.2",
                    "samtools": 1.11,
                    "pigz": "2.3.4"
                },
                "BOWTIE2_ASSEMBLY_BUILD": {
                    "bowtie2": "2.4.2"
                },
                "BOWTIE2_HOST_REMOVAL_ALIGN": {
                    "bowtie2": "2.4.2"
                },
                "BOWTIE2_HOST_REMOVAL_BUILD": {
                    "bowtie2": "2.4.2"
                },
                "BOWTIE2_PHIX_REMOVAL_ALIGN": {
                    "bowtie2": "2.4.2"
                },
                "BOWTIE2_PHIX_REMOVAL_BUILD": {
                    "bowtie2": "2.4.2"
                },
                "CATPACK_ADDNAMES_BINS": {
                    "catpack": 6.0
                },
                "CATPACK_BINS": {
                    "catpack": 6.0
                },
                "CATPACK_SUMMARISE_BINS": {
                    "catpack": 6.0
                },
                "CAT_DB_UNTAR": {
                    "untar": 1.34
                },
                "CHECKM_LINEAGEWF": {
                    "checkm": "1.2.3"
                },
                "CHECKM_QA": {
                    "checkm": "1.2.3"
                },
                "CHECKM_UNTAR": {
                    "untar": 1.34
                },
                "CONCAT_BINQC_TSV": {
                    "csvtk": "0.31.0"
                },
                "CONCOCT_CONCOCT": {
                    "concoct": "1.1.0"
                },
                "CONCOCT_CONCOCTCOVERAGETABLE": {
                    "concoct": "1.1.0"
                },
                "CONCOCT_CUTUPFASTA": {
                    "concoct": "1.1.0"
                },
                "CONCOCT_EXTRACTFASTABINS": {
                    "concoct": "1.1.0"
                },
                "CONCOCT_MERGECUTUPCLUSTERING": {
                    "concoct": "1.1.0"
                },
                "CONVERT_DEPTHS": {
                    "bioawk": 20110810
                },
                "FAIDX": {
                    "samtools": 1.21
                },
                "FASTQC_RAW": {
                    "fastqc": "0.12.1"
                },
                "FASTQC_TRIMMED": {
                    "fastqc": "0.12.1"
                },
                "FREEBAYES": {
                    "freebayes": "1.3.6"
                },
                "GENOMAD_ENDTOEND": {
                    "genomad": "1.5.2"
                },
                "GENOMAD_UNTAR": {
                    "untar": 1.34
                },
                "GTDBTK_DB_PREPARATION": {
                    "tar": 1.3
                },
                "GTDBTK_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "GUNZIP_BINS": {
                    "gunzip": 1.13
                },
                "GUNZIP_SHORTREAD_ASSEMBLIES": {
                    "gunzip": 1.13
                },
                "MAG_DEPTHS": {
                    "python": "3.6.7",
                    "pandas": "1.1.5"
                },
                "MAG_DEPTHS_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "MAXBIN2": {
                    "maxbin2": "2.2.7"
                },
                "MEGAHIT": {
                    "megahit": "1.2.9"
                },
                "METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS_SHORTREAD": {
                    "metabat2": 2.15
                },
                "METABAT2_METABAT2": {
                    "metabat2": 2.17
                },
                "PRODIGAL": {
                    "prodigal": "2.6.3",
                    "pigz": 2.6
                },
                "PROKKA": {
                    "prokka": "1.14.6"
                },
                "PYDAMAGE_ANALYZE": {
                    "pydamage": 1.0
                },
                "PYDAMAGE_FILTER": {
                    "pydamage": 1.0
                },
                "QUAST": {
                    "python": "3.7.6",
                    "metaquast": "5.0.2"
                },
                "QUAST_BINS": {
                    "python": "3.7.6",
                    "metaquast": "5.0.2"
                },
                "QUAST_BINS_SUMMARY": {
                    "sed": 4.7
                },
                "SEQKIT_STATS": {
                    "seqkit": "2.9.0"
                },
                "SPLIT_FASTA": {
                    "python": "3.6.7",
                    "biopython": "1.7.4",
                    "pandas": "1.1.5"
                },
                "Workflow": {
                    "nf-core/mag": "v4.1.0dev"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-25T16:34:16.949282271"
    }
}